File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()== 0) return 0;
         
        sort(nums.begin(),nums.end());
        
        int count=1;
        int max1=1;
        for(int i =0;i<nums.size()-1;i++){
            if(nums[i+1]-nums[i]==1) count++;
            else if(nums[i+1]-nums[i]!=0) count=1;
            max1 =max(max1,count);
        }
        return max1;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> ele = new HashSet<>();
       for (int num : nums) ele.add(num);
       int maxi=0;
       for(int num:ele){
        if(!ele.contains(num-1)){
            int current =num;
            int count=1;
            while(ele.contains(current+1)){
                current ++;
                count++;
            }
            maxi = Math.max(maxi,count);
        }
       }
       return maxi;
    }
}

